torch/_inductor/ir.py

```diff
@@ -4410,6 +4416,7 @@ def canonicalize(self):
         sizes = self.get_size()
         strides = self.get_stride()
         strides = [sizevars.size_hint(x) for x in strides]
+        # TODO: I can't tell if the symbols here are temporary
```

Comment:
assert that they are not part of `index` at the end of the function?
They look temporary to me, as they are all replaced in the `sympy_subs` below. The interesting thing is that they are replaced by a new variable `ci` which does escape this function...

Reply:
I'm going to do the asserts separately as they're likely to fail